starknet_py/contract.py

```diff
@@ -338,7 +356,70 @@ async def call(
         result = await self.call_raw(block_hash=block_hash, block_number=block_number)
         return self._payload_transformer.deserialize(result)
 
-    # TODO (#1182): add `cairo_version` parameter
+
+@add_sync_methods
+@dataclass
+class PreparedFunctionInvoke(ABC, PreparedCallBase):
+    _contract_data: ContractData
+    _account: Optional[BaseAccount]
+
+    def __post_init__(self):
+        if self._account is None:
+            raise ValueError(
+                "Contract instance was created without providing an Account. "
+                "It is not possible to prepare and send an invoke transaction."
+            )
+
+    @property
+    def get_account(self):
+        if self._account is not None:
+            return self._account
+
+        raise ValueError(
+            "The account is not defined. It is not possible to send an invoke transaction."
+        )
```

Comment:
1. Do we really want to expose account there? It like a bit confusing property to have in `PreparedFunctionInvokeVX`, since it's only used as an internal helper and user would probably be aware which account they used to create prepared invoke anyway.
2. Also, could we maybe make `_account` non-optional? We can check whether it's `None` in `prepare_invoke_vX` before creating instance `PreparedFunctionInvokeVX`.

Reply:
1. The main motivation here is to avoid pyright errors when using `_account`. We are sure that it isn't `None`, as we make this verification in `__post_init__` (excluding cases where a user might change it manually on an existing `PreparedFunctionInvokeVX` instance). Considering that an account is a prerequisite for transaction invocation, it kind of makes sense to have such a property in `PreparedFunctionInvoke`. As an alternative, I could make `_get_account` internal.
2. The `account` in the `Contract` class might be `None`, resulting in a None when calling prepare_invoke_vX. We can potentially make this assertion there, but then we would need to add this in four places instead of just one.

Reply:
> As an alternative, I could make _get_account internal.

That is one solution, because it's overly clear what it does in the context of `PreparedFunctionInvokeVX`

> We can potentially make this assertion there, but then we would need to add this in four places instead of just one.

More like 4 instead of 2. Also, can we maybe check `_account` right away when it's passed to constructor?

Reply:
> More like 4 instead of 2

It is implemented now only within the abstract `PreparedFunctionInvoke` class

> can we maybe check _account right away when it's passed to constructor?

This is a `dataclass`, so the only option is `__post_init__`